src/app/header/shared/utils.ts

```diff
@@ -0,0 +1,49 @@
+import dayjs from 'dayjs';
+
+export function getButtonTerms(date: dayjs.Dayjs = dayjs()): string[] {
```

Comment:
The only difference between these two functions is the final output. I would maybe suggest writing a format function and try and re-use a lot of the logic (checking the ranges for month etc)